tests/test_dataset_cards.py

```diff
@@ -0,0 +1,104 @@
+# coding=utf-8
+# Copyright 2021 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+from subprocess import check_output
+from typing import List
+
+import pytest
+
+from datasets.utils.logging import get_logger
+from datasets.utils.metadata import DatasetMetadata
+from datasets.utils.readme import ReadMe
+
+from .utils import slow
+
+
+repo_path = Path.cwd()
+
+logger = get_logger(__name__)
+
+
+def get_changed_datasets(repo_path: Path) -> List[Path]:
+    diff_output = check_output(["git", "diff", "--name-only", "HEAD..origin/master"], cwd=repo_path)
+    changed_files = [Path(repo_path, f) for f in diff_output.decode().splitlines()]
+    changed_datasets_unique = []
+    changed_datasets = [
+        f.parent.parts[-1]
+        for f in changed_files
+        if f.exists() and f.parent.parent.name == "datasets" and f.parent.parent.parent.name == "datasets"
+    ]
+
+    for dataset_name in changed_datasets:
+        if dataset_name not in changed_datasets_unique:
+            changed_datasets_unique.append(dataset_name)
+
+    return changed_datasets_unique
```

Comment:
I think your `get_changed_datasets` function has some issues, here is a fix:

```suggestion
def get_changed_datasets(repo_path: Path) -> List[Path]:
    diff_output = check_output(["git", "diff", "--name-only", "HEAD..origin/master"], cwd=repo_path)
    changed_files = [Path(repo_path, f) for f in diff_output.decode().splitlines()]
    changed_datasets_unique = set(
        f.parent.parts[-1]
        for f in changed_files
        if f.exists() and f.name.lower() == "readme.md" and f.parent.parent.name == "datasets"
    )
    return sorted(changed_datasets_unique)
```

Basically this test never passed:
```python
f.parent.parent.name == "datasets" and f.parent.parent.parent.name == "datasets"
```

Reply:
Also don't forget to merge master into this branch to get the latest changes ;)

Reply:
@lhoestq Don't we want to look for the changes in `datasets/datasets/xyz`?

And the card's parent is xyz, parent.parent is datasets, and parent.parent.parent is datasets?

Reply:
parent.parent.parent may be whatever users want.